### PR TITLE
AzureMonitor: Traces - Testing

### LINKED ISSUE #63123: AzureMonitor: Traces - Testing
E2E and unit tests should be included testing all aspects of the Traces view.

Depends on:
#63117 
#63118 
#63119 
#63120 
#63121 
#63124

### PR DESCRIPTION
**Pull request by aangelisc**
This PR adds testing to the existing Traces work. This includes frontend, backend, and e2e testing. 

- Fix bug where the correct `queryType` wasn't selected, preventing template variables from being interpolated.
- Fix/update existing tests.
- Ensure test selectors are consistent.
- Added `TracesQueryEditor` test.
- Added `FormatAsField` test.
- Added `TraceTypeField` test.
- Added `Filters` test.
- Fixed multi-selection on `Filters`.
- Updated backend datasource test.
- Updated backend response-table-frame test.
- Updated backend macros test.
- Added relevant test data.
- Added generation for test data using `CheckGoldenJSONResponse` util.
- Updated all tests and removed old `expectedFrame` properties.

Part of #63117, #63127, #63124, #63118, #63121, #63119, #63129
Fixes #63123

### CODE REVIEW

**Comment by aangelisc** [comment]
@JoaoSilvaGrafana I had to add another property to get my tests to work because of the way the `react-dom` works in tests (all DOMs are touch capable by definition which breaks my tests). Let me know if this is a problem, I'm happy to work with you on reducing the number of props if needed.

File: .betterer.results [hunk 1]
```diff
@@ -3816,8 +3816,7 @@ exports[`better eslint`] = {
     "public/app/plugins/datasource/azuremonitor/azure_log_analytics/azure_log_analytics_datasource.test.ts:5381": [
       [0, 0, 0, "Unexpected any. Specify a different type.", "0"],
       [0, 0, 0, "Unexpected any. Specify a different type.", "1"],
-      [0, 0, 0, "Unexpected any. Specify a different type.", "2"],
-      [0, 0, 0, "Unexpected any. Specify a different type.", "3"]
```
Comment (joshhunt) [inline .betterer.results:3820]: 🚀

**Comment by JoaoSilvaGrafana** [comment]
> @JoaoSilvaGrafana I had to add another property to get my tests to work because of the way the `react-dom` works in tests (all DOMs are touch capable by definition which breaks my tests). Let me know if this is a problem, I'm happy to work with you on reducing the number of props if needed.

As discussed, not ideal to keep adding props but we will revisit this on the Select component rework so its ok for now 👍

**Review by asimpson — APPROVED** [review]
Whew, nicely done on this! :trophy:

File: pkg/tsdb/azuremonitor/loganalytics/azure-response-table-frame_test.go [hunk 2]
```diff
@@ -193,7 +194,418 @@ func TestLogTableToFrame(t *testing.T) {
...
+	tests := []struct {
+		name          string
+		testFile      string
+		expectedFrame func() *data.Frame
+		resultFormat  dataquery.ResultFormat
+	}{
+		{
+			name:     "multi trace",
+			testFile: "traces/1-traces-multiple-table.json",
+			expectedFrame: func() *data.Frame {
```
Comment (bossinc) [inline pkg/tsdb/azuremonitor/loganalytics/azure-response-table-frame_test.go:217]: This looks like a good use case for `CheckGoldenJSONResponse`. Here is an example of a helper function. https://github.com/grafana/github-datasource/blob/main/pkg/testutil/frames.go#L21

**Review by bossinc — COMMENTED** [review]
so much testing! i love it

File: pkg/tsdb/azuremonitor/loganalytics/azure-response-table-frame_test.go (hunk 2 above)
Comment (aangelisc) [reply pkg/tsdb/azuremonitor/loganalytics/azure-response-table-frame_test.go:217]: I'll have a look at this 👀

Comment (aangelisc) [reply pkg/tsdb/azuremonitor/loganalytics/azure-response-table-frame_test.go]: Thank you @bossinc for the great recommendation! That's a bunch of redundant code removed 😊

**Pull request merged**